nautobot/docs/models/dcim/deviceredundancygroups.md

```diff
@@ -0,0 +1,250 @@
+
+This document provides an example of generating a Cisco ASA device's desired failover configuration based on data stored in Nautobot Device Redundancy Groups. 
+A GraphQL query is used to retrieve the relevant data, which is then rendered through a Jinja2 template to produce the desired configuration.
+
+As one of the use cases for Device Redundacy Groups introduced in Nautobot 1.5 release is to model failover pairs, we will use the ASA 5515 in this example with details as follows:
+
+* Our firewall failover pair is composed of two devices named "nyc-fw-primary" and "nyc-fw-secondary"
+* Each firewall device forming a failover cluster has a virtual interface dedicated for failover usage (named "failover-link") and addressed in 172.27.48.0/31 network
+* Virtual interface has a parent interface assigned ("gigabitethernet0/3")
+
+### Querying for the data
+
+To retrieve information about devices forming an ASA Failover pair, we will use GraphQL query and the `get_gql_failover_details` Python method.
+This method takes a `device_name` as an argument.
+
+```python
+import pynautobot
+import json
+
+query = """
+query ($device_name: [String]) {
+    devices(name__ie: $device_name) {
+        name
+        device_redundancy_group {
+            name
+            members {
+                name
+                device_redundancy_group_priority
+                interfaces(name__ie: "failover-link") {
+                    type
+                    name
+                    ip_addresses {
+                        host
+                        prefix_length
+                    }
+                    parent_interface {
+                        name
+                        type
+                    }
+                }
+            }
+        }
+    }
+}
+"""
+
+def get_gql_failover_details(device_name):
+    variables = {"device_name": device_name}
+    nb = pynautobot.api(
+        url="http://localhost:8080",
+        token="aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    )
+    return nb.graphql.query(query=query, variables=variables)
+```
+
+### Retrieving the data - Primary Failover Unit ("nyc-fw-primary")
+
+An example data returned from Nautobot is presented below.
+
+```python
+>>> primary_unit_details = get_gql_failover_details("nyc-fw-primary").json
+```
+
+```json
+{'data': {'devices': [{'name': 'nyc-fw-primary', 'device_redundancy_group': {'name': 'nyc-firewalls', 'members': [{'name': 'nyc-fw-primary', 'device_redundancy_group_priority': 100, 'interfaces': [{'type': 'VIRTUAL', 'name': 'failover-link', 'ip_addresses': [{'host': '172.27.48.0', 'prefix_length': 31}], 'parent_interface': {'name': 'gigabitethernet0/3', 'type': 'A_1000BASE_T'}}]}, {'name': 'nyc-fw-secondary', 'device_redundancy_group_priority': 50, 'interfaces': [{'type': 'VIRTUAL', 'name': 'failover-link', 'ip_addresses': [{'host': '172.27.48.1', 'prefix_length': 31}], 'parent_interface': {'name': 'gigabitethernet0/3', 'type': 'A_1000BASE_T'}}]}]}}]}}
```

Comment:
Can we use `json.dumps` or similar here and below so that the JSON is more human-readable?